File: problemset/House_Robber.cpp
class Solution {
public:
    int dp[100010][2];
    int rob(vector<int>& nums) {
        memset(dp, 0, sizeof dp);
        for (int i = 0; i < nums.size(); i++) {
            if (!i) {
                dp[i][1] = nums[i];
                continue;
            }
            dp[i][0] = max(dp[i-1][0], dp[i-1][1]);
            dp[i][1] = nums[i] + dp[i-1][0];
        }
        return max(dp[nums.size()-1][0], dp[nums.size()-1][1]);
    }
};

File: activity/20200904/RabbitLee/House_Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 0:
            return 0
        dp = [[0, 0] for i in range(n)]
        dp[0] = [nums[0], 0]
        for i in range(1, n):
            dp[i][0] = dp[i - 1][1] + nums[i]
            dp[i][1] = max(dp[i - 1])
        return max(dp[n - 1])

